lib/posix/sched.c

```diff
@@ -41,3 +42,57 @@ int sched_get_priority_max(int policy)
 	errno = EINVAL;
 	return -1;
 }
+
+/**
+ * @brief Get scheduling parameters
+ *
+ * See IEEE 1003.1
+ */
+int sched_getparam(pid_t pid, struct sched_param *param)
+{
+	/* TODO: Ensure the Zephyr POSIX process id (PID) value is assigned
+	 * with its first thread's id (TID).
+	 */
+	struct sched_param dummy_param = {0};
+	pthread_t tid = (pthread_t)pid;
+	int t_policy = -1;
+	int ret = -1;
+
+	if (param == NULL) {
+		param = &dummy_param;
+	}
+
+	if (tid == 0) {
+		tid = pthread_self();
+	}
+
+	ret = pthread_getschedparam(tid, &t_policy, param);
+	errno = ret;
+
+	return ((ret) ? -1 : 0);
+}
+
+/**
+ * @brief Get scheduling policy
+ *
+ * See IEEE 1003.1
+ */
+int sched_getscheduler(pid_t pid)
+{
+	/* TODO: Ensure the Zephyr POSIX process id (PID) value is assigned
+	 * with its first thread's id (TID).
+	 */
+	struct sched_param dummy_param = {0};
+	pthread_t tid = (pthread_t)pid;
```

Comment:
It would be good to get some additional input here. Personally, I think that pid_t and pthread_t should not be used interchangeably.

Some current plans for pid_t can be gleaned from #65920. If you're on Discord, I would be happy to chat a bit about it

Reply:
I've changed to an inline function call, which is rather a placeholder for this PR till the design decision on PID implementation.

https://github.com/zephyrproject-rtos/zephyr/pull/67028/files#diff-549102eb96e194323c2cf4a5363fe6a41aecb1978080e0117c7027f91391b298R108-R115

Reply:
Sure - well, the design decision is that Zephyr currently runs in a way that is approximately equivalent to a "single process" environment. So there would be one (maybe two at most) PIDs.

PIDs and TIDs are not considered to be equivalent. This is particularly important because processes do not share resources (by default) and have necessarily disjoint address spaces (even if not virtualized) while all threads belonging to a particular process would.

So it's possible to get the PID associated with the currently running thread by simply calling `getpid()`, but there is no simple numerical mapping between TIDs and PIDs (at this time).

That would really also necessarily be something that would need to be implemented at the kernel level, and not in the POSIX library.